src/coreclr/jit/lowerxarch.cpp

```diff
@@ -241,6 +241,54 @@ GenTree* Lowering::LowerMul(GenTreeOp* mul)
     return mul->gtNext;
 }
 
+//----------------------------------------------------------------------------------------------
+// Lowering::TryLowerAndNegativeOne:
+//    If safe, lowers a tree AND(X, CNS(-1)) to X.
+//
+// Arguments:
+//    node - GT_AND node of integral type
+//
+// Return Value:
+//    Returns the replacement node if one is created else nullptr indicating no replacement
+GenTree* Lowering::TryLowerAndNegativeOne(GenTreeOp* node)
+{
+    assert(node->OperIs(GT_AND));
+
+    if (!varTypeIsIntegral(node))
+        return nullptr;
+
+    if (node->gtSetFlags())
+        return nullptr;
+
+    GenTree* op2 = node->gtGetOp2();
+
+    if (!op2->IsIntegralConst(-1))
+        return nullptr;
+
+#ifdef TARGET_64BIT
+    if (!node->TypeIs(TYP_LONG) || !op2->TypeIs(TYP_LONG))
+        return nullptr;
+#else  // TARGET_64BIT
+    assert(op2->TypeIs(TYP_INT));
+
+    if (!node->TypeIs(TYP_INT))
+        return nullptr;
+#endif // !TARGET_64BIT
```

Comment:
I still don't understand why we wouldn't be able to do this for `TYP_INT` on 64-bit. Can you share an example please?

Reply:
I will try to enable it again and run Fuzzlyn.

Reply:
Thought about it some more, and I think you might be right. Whatever is using the result of `AND(VAR, -1)` shouldn't be caring about the upper 32bits anyway; therefore, it should be fine to just return `VAR`.